test/renderscript/jit_copy.cpp

```diff
@@ -0,0 +1,178 @@
+#include "Halide.h"
+
+using namespace Halide;
+using namespace Halide::Internal;
+
+// This visitor takes a string snapshot of all Pipeline IR nodes.
+class SnapshotPipeline : public IRMutator {
+    void visit(const Pipeline *op) {
+        pipeline << op->produce;
+    }
+
+public:
+    std::ostringstream pipeline;
+};
+
+Image<uint8_t> make_interleaved_image(uint8_t *host, int W, int H, int nChannels) {
+    buffer_t buf = { 0 };
+    buf.host = host;
+    buf.extent[0] = W;
+    buf.stride[0] = nChannels;
+    buf.extent[1] = H;
+    buf.stride[1] = buf.stride[0] * buf.extent[0];
+    buf.extent[2] = nChannels;
+    buf.stride[2] = 1;
+    buf.elem_size = 1;
+    return Image<uint8_t>(&buf);
+}
+
+Image<uint8_t> make_planar_image(uint8_t *host, int W, int H, int nChannels) {
+    buffer_t buf = { 0 };
+    buf.host = host;
+    buf.extent[0] = W;
+    buf.stride[0] = 1;
+    buf.extent[1] = H;
+    buf.stride[1] = buf.stride[0] * buf.extent[0];
+    buf.extent[2] = nChannels;
+    buf.stride[2] = buf.stride[1] * buf.extent[1];
+    buf.elem_size = 1;
+    return Image<uint8_t>(&buf);
+}
+
+std::string copy_interleaved(bool isVectorized = false, int nChannels = 4) {
+    ImageParam input8(UInt(8), 3, "input");
+    input8.set_stride(0, nChannels)
+        .set_stride(1, Halide::Expr())
+        .set_stride(2, 1)
+        .set_bounds(2, 0, nChannels);  // expecting interleaved image
+    uint8_t in_buf[128 * 128 * nChannels];
+    uint8_t out_buf[128 * 128 * nChannels];
+    Image<uint8_t> in = make_interleaved_image(in_buf, 128, 128, nChannels);
+    Image<uint8_t> out = make_interleaved_image(out_buf, 128, 128, nChannels);
+    input8.set(in);
+
+    Var x, y, c;
+    Func result("result");
+    result(x, y, c) = input8(x, y, c);
+    result.output_buffer()
+        .set_stride(0, nChannels)
+        .set_stride(1, Halide::Expr())
+        .set_stride(2, 1)
+        .set_bounds(2, 0, nChannels);  // expecting interleaved image
+
+    result.bound(c, 0, nChannels);
+    result.image(x, y, c, DeviceAPI::Renderscript);
+    if (isVectorized) result.vectorize(c);
+
+    auto p = new SnapshotPipeline();
+    result.add_custom_lowering_pass(p);
+    result.realize(out);
+
+    return p->pipeline.str();
+}
+
+std::string copy_interleaved_vectorized(int nChannels = 4) {
+    return copy_interleaved(true, nChannels);
+}
+
+int main(int argc, char **argv) {
+    std::string expected_vectorized_ir =
+        R"|(let copy_to_device_result$2 = halide_copy_to_device(result.buffer, halide_renderscript_device_interface())
```

Comment:
You could try using IRMatch with wildcards for the bits you don't care about instead of a string literal. That might be more invariant to minor changes in, e.g., IR printing.